scene/gui/file_dialog.cpp

```diff
@@ -596,7 +597,7 @@ void FileDialog::set_current_file(const String &p_file) {
 	int lp = p_file.find_last(".");
 	if (lp != -1) {
 		file->select(0, lp);
-		if (file->is_inside_tree())
+		if (file->is_inside_tree() && !Engine::get_singleton()->is_editor_hint())
```

Comment:
I'm not sure this is the right fix, it would break the behaviour of `FileDialog` when used in the editor itself (though the editor mainly uses `EditorFileDialog`).

It seems a bit dangerous to me that we execute all this logic simply when changing the initial configuration of the `FileDialog` via the inspector.

Reply:
Sorry, I didn't think about that... So what would be a proper fix for this ?

Reply:
I don't know for now, would like some input from @reduz.

Reply:
This makes me think of https://github.com/godotengine/godot/issues/17592, where sometimes you don't want `tool` logic to run when inside the edited scene root.

Reply:
I think this is because the intended behaviour is, if you set the current file to something like "aaa.txt", that the focus will be given to the LineEdit with the file name with "aaa" selected so that the user can easily change the filename without erasing the extension (before saving for example).

Reply:
Discussed this with @reduz on IRC, the correct fix would be to use `file->is_inside_tree() && get_tree()->is_node_being_edited(file)`.

Reply:
Tested locally and confirmed working, so I amended the commit accordingly.